vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,902 @@
+list). Thus `ignore_event_order` and `tmp_event_nr_var` are not specified.
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(rsp_y_imwg, cb_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CB",
+      PARAM = "IMWG Clinical Benefit by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSORRES, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD == "CB" & AVALC == "Y"
+)
+```
+
+Similarly, we can define the parameters:
+
+* CR or better response (`PARAMCD` = `CR`),  
+* VGPR or better response (`PARAMCD` =`VGPR`).
+
+```{r}
+cr_y_imwg <- event(
+  description = "Define sCR or CR as response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("sCR", "CR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(cr_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CRRSP",
+      PARAM = "IMWG Complete Response by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+
+vgpr_y_imwg <- event(
+  description = "Define sCR, CR or VGPR as response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("sCR", "CR", "VGPR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(vgpr_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "VGPRRSP",
+      PARAM = "IMWG VGPR Rate by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSORRES, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD %in% c("CRRSP", "VGPRRSP") & AVALC == "Y"
+)
+```
+
+## Derive Best Overall Response Parameter {#bor}
+
+The function `admiral::derive_extreme_event()` can be used to derive the best
+overall response (without confirmation required) parameter.
+
+Please note that the order of the events specified for `events` is important.
+For example, a subject with `PR`, `PR`, `CR` qualifies for both `bor_cr` and
```

Comment:
in IMWG we don't need unconfirmed response, its always confirmed. we need to create CBOR, CCB etc parameter as per workflow correct? this section seems like from RECIST1.1, we should update accoridingly

Reply:
I think its good to add CBOR as and PARAM as mentioned "confirmed" word. It will help to be explicit and consistent with our other Vignettes.

Reply:
I think our confirmation has a slightly different meaning than the one in Recist criteria. There we defined the time we asked for the response to be sustained. Here we are only interested in what happens on the next visit. Perhaps it is we should not mention confirmation in names?

Reply:
updated